src/IdentityServer4/src/Hosting/IdentityServerMiddleware.cs

```diff
@@ -51,7 +52,9 @@ public async Task Invoke(HttpContext context, IEndpointRouter router, IUserSessi
                 {
                     _logger.LogInformation("Invoking IdentityServer endpoint: {endpointType} for {url}", endpoint.GetType().FullName, context.Request.Path.ToString());
 
-                    var result = await endpoint.ProcessAsync(context);
+                    var cancellationToken = context.Features.Get<IHttpRequestLifetimeFeature>()?.RequestAborted ?? default;
```

Comment:
The ability for **RequestAborted** appears to be the main capability added. Isn't there a way to provide access through just the middleware without passing in cancellationtoken everywhere?  Also, I don't see any tests for possible insecure states cancellation may cause.

Reply:
Thank for your feedback. It's not possible to use a middleware in this case and it's important to propagate the cancellation token to the lowest layers to cancel all operations (sql queries, http calls etc.)